mods/gennext/abilities.js

```diff
@@ -588,18 +588,20 @@ exports.BattleAbilities = {
 			pokemon.addVolatile('shadowtag');
 		},
 		effect: {
+			duration: 2,
 			onFoeModifyPokemon: function (pokemon) {
 				if (pokemon.ability !== 'shadowtag') {
 					pokemon.tryTrap(true);
 				}
 			}
 		},
+		onBeforeMovePriority: 15,
```

Comment:
Shadow Tag on the switch-in might last for 2 turns if we just keep the `duration: 2`. (Untested since I never considered having only the duration, but I'm going off how Drizzle works and whatnot).